crates/parcel/src/requests/asset_graph_request.rs

```diff
@@ -0,0 +1,235 @@
+use std::collections::{HashMap, HashSet};
+use std::sync::mpsc::channel;
+use std::sync::Arc;
+
+use parcel_core::asset_graph::{AssetGraph, DependencyNode, DependencyState};
+use parcel_core::types::Dependency;
+use pathdiff::diff_paths;
+use petgraph::graph::NodeIndex;
+
+use crate::request_tracker::{
+  Request, RequestId, ResultAndInvalidations, RunRequestContext, RunRequestError,
+};
+
+use super::asset_request::{AssetRequest, AssetRequestOutput};
+use super::entry_request::{EntryRequest, EntryRequestOutput};
+use super::path_request::{PathRequest, PathRequestOutput};
+use super::target_request::{TargetRequest, TargetRequestOutput};
+use super::RequestResult;
+
+/// The AssetGraphRequest is in charge of building the AssetGraphRequest
+/// In doing so, it kicks of the TargetRequest, PathRequest and AssetRequests.
+#[derive(Debug, Hash)]
+pub struct AssetGraphRequest {}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct AssetGraphRequestOutput {
+  pub graph: AssetGraph,
+}
+
+impl Request for AssetGraphRequest {
+  fn run(
+    &self,
+    mut request_context: RunRequestContext,
+  ) -> Result<ResultAndInvalidations, RunRequestError> {
+    let mut graph = AssetGraph::new();
+    let (tx, rx) = channel();
+    // TODO: Move this out later
```

Comment:
move to context? where is it moving?

Reply:
I updated the comment. We were thinking the work count tracking might be better off as part of the `request_context.queue_request` API.